lib/crypto/base36.js

```diff
@@ -0,0 +1,30 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+var crypto = require('crypto')
+
+function base36(len) {
+  var out = ''
+  while (out.length < len) {
+    var rand = crypto.randomBytes(len)
```

Comment:
It sounds scary, but if you do a search of 'randomBytes' on this repo, you'll see that we use synchronous calls for a bunch of things (non-exhaustive list: emailCode, authSalt, tokenVerificationId, password forgot passCode, ...). I believe the key point in here is this part:

> The only time when generating the random bytes may conceivably block for a longer period of time is right after boot, when the whole system is still low on entropy.

May be worth a follow up bug "async all the randomBytes", or it might not be an issue in practice.